### DISCUSSION TITLE
Menu doesn't contain recently closed tab

### DISCUSSION THREAD

**Discussion by lonix1** [M-Reimer/undoclosetab#130] [General]
It's often the case that the restore menu does not contain a tab which I closed recently, but contains other tabs, even those I closed long ago.

I can't reproduce the bug, but it's definitely there.

Has anyone else encountered this?

(I'm using the latest version of everything.)

**Comment by M-Reimer** [comment]
Please start with trying this: https://github.com/M-Reimer/undoclosetab/wiki/How-to-fix-broken-browser-session

There are other issues but all worked around "somewhat well" with workarounds. In fact creating a "somewhat usable" tab restoring Add-on is made really difficult by Mozilla. In theory (according to API documentation) it should be possible with just one line of code. But thanks to Mozilla it needs around 100 more lines just with workarounds of various bugs.

**Reply by lonix1** [reply]
Thanks. I doubt that's the issue in this case because it does work, it just does not have a specific tab. But I can continue using it and it shows other recent tabs. Like I wrote above, I have no idea how to reproduce it. Sometimes it just "forgets" a specific tab for no reason I can determine.

Regarding the workarounds, I saw the large number of "upstream" tagged issues. Must be frustrating for you to maintain so many workarounds!

Thanks for maintaining this extension, it's very useful despite these issues.

**Reply by lonix1** [reply]
I found a reproduction.

When I close any page on "https://getbootstrap.com" it shows in the menu. But after a few other closes it moves down in the list then disappears. Even though there are even older items below it.

I've seen this with other sites too.
